runtimes/sklearn/mlserver_sklearn/sklearn.py

```diff
@@ -60,15 +56,38 @@ def _check_request(self, payload: types.InferenceRequest) -> types.InferenceRequ
                         f"({request_output.name} was received)"
                     )
 
+        # Regression models do not support `predict_proba`
+        if PREDICT_PROBA_OUTPUT in [o.name for o in payload.outputs]:
+            # Ensure model supports it
+            maybe_regressor = self._model
+            if isinstance(self._model, Pipeline):
+                maybe_regressor = maybe_regressor.steps[-1][-1]
+
+            if not hasattr(maybe_regressor, PREDICT_PROBA_OUTPUT):
+                raise InferenceError(
+                    f"{type(maybe_regressor)} models do not support '{PREDICT_PROBA_OUTPUT}"
+                )
+
         return payload
 
     def _predict_outputs(
         self, payload: types.InferenceRequest
     ) -> List[types.ResponseOutput]:
-        model_input = payload.inputs[0]
-
         default_codec = NumpyCodec()
-        input_data = self.decode(model_input, default_codec=default_codec)
+        # TODO: how to set default codec here? Needs to be in the request...?
+        decoded_request = self.decode_request(payload)
```

Comment:
Good point! The `decode_request()` method doesn't support a default codec atm. Maybe we should extend that method so that it does something similar to what the `decode()` method does? Happy to leave that out of the scope of the PR though.